planner/core/task.go

```diff
@@ -711,6 +711,14 @@ func finishCopTask(ctx sessionctx.Context, task task) task {
 	if !ok {
 		return task
 	}
+	if t.tablePlan != nil {
+		tp := t.tablePlan
+		for len(tp.Children()) > 0 {
+			tp = tp.Children()[0]
```

Comment:
```suggestion
			if len(tp.Children()) == 1 {
				tp = tp.Children()[0]
			} else {
				join := tp.(*PhysicalBroadCastJoin)
				tp = join.children[1-join.InnerChildIdx]
			}
```

We should consider broadcast join in cop task. And by the way, this check can be moved to https://github.com/pingcap/tidb/pull/18516/files#diff-2f445d8cb9c2f7d8ec4fb0586ae1f510R730